week-2/errors/0.js

```diff
@@ -3,7 +3,25 @@
 // then call the function capitalise with a string input
 // interpret the error message and figure out why it's happening, if your prediction was wrong
 
-function capitalise(str) {
+/* function capitalise(str) {
   let str = `${str[0].toUpperCase()}${str.slice(1)}`;
   return str;
 }
+
+
+function, named capitalise, is meant to take a text input and change its first letter to a capital letter.
+It takes a text (str) as an input parameter.
+It converts the first letter of the input text to uppercase using str[0].toUpperCase().
+t takes the remaining part of the text after the first letter using str.slice(1).
+It combines the uppercase first letter with the rest of the text to create a new text string where the first letter is capitalized.
+ Finally, it returns this newly formed text string.
+
+ The problem is that the code tries to use the name "str" for two different things inside the function, which isn't allowed in JavaScript. we can't have two variables with the same name within the same block of code.
+  To solve this issue, just rename the second "str" to something else, like "modifiedStr": */
+
+function capitalise(str) {
+  Str = `${str[0].toUpperCase()}${str.slice(1)}`;
+  return Str;
+}
```

Comment:
Your explanation is correct here 👍 
But I think your implementation may be improved.

Firstly, in JS we have a convention to use `camelCasedNames`, it means your new variable name should start with lowercased latter. Your suggestion t use `modifiedStr` is perfect for this purpose!

Secondly, if you introduce a new variable, you should use either `let` or `const` in the beginning.

Thirdly, just `str = `${str[0].toUpperCase()}${str.slice(1)}`` (without let in front of it) should work too. As all the function arguments (like `str` here) may be changed as any variable declared with `let`.
But we should be careful with this approach, because after changing initial value of `str` we won't be able to refer to initial value any more.

Fourthly, you may even avoid creating variables her and just:
```
return `${str[0].toUpperCase()}${str.slice(1)}`;
```